eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version0/ChannelCodecs0.scala

```diff
@@ -165,7 +173,36 @@ private[channel] object ChannelCodecs0 {
         ("txid" | bytes32) ::
         ("remotePerCommitmentPoint" | publicKey)).as[RemoteCommit].decodeOnly
 
-    val updateMessageCodec: Codec[UpdateMessage] = lightningMessageCodec.narrow(f => Attempt.successful(f.asInstanceOf[UpdateMessage]), g => g)
+    val updateFulfillHtlcCodec: Codec[UpdateFulfillHtlc] = (
+      ("channelId" | bytes32) ::
+        ("id" | uint64overflow) ::
+        ("paymentPreimage" | bytes32) ::
+        ("tlvStream" | provide(TlvStream.empty[UpdateFulfillHtlcTlv]))).as[UpdateFulfillHtlc]
+
+    val updateFailHtlcCodec: Codec[UpdateFailHtlc] = (
+      ("channelId" | bytes32) ::
+        ("id" | uint64overflow) ::
+        ("reason" | varsizebinarydata) ::
+        ("tlvStream" | provide(TlvStream.empty[UpdateFailHtlcTlv]))).as[UpdateFailHtlc]
+
+    val updateFailMalformedHtlcCodec: Codec[UpdateFailMalformedHtlc] = (
+      ("channelId" | bytes32) ::
+        ("id" | uint64overflow) ::
+        ("onionHash" | bytes32) ::
+        ("failureCode" | uint16) ::
+        ("tlvStream" | provide(TlvStream.empty[UpdateFailMalformedHtlcTlv]))).as[UpdateFailMalformedHtlc]
+
+    val updateFeeCodec: Codec[UpdateFee] = (
+      ("channelId" | bytes32) ::
+        ("feeratePerKw" | feeratePerKw) ::
+        ("tlvStream" | provide(TlvStream.empty[UpdateFeeTlv]))).as[UpdateFee]
+
+    val updateMessageCodec: Codec[UpdateMessage] = discriminated[UpdateMessage].by(uint16)
+      .typecase(128, updateAddHtlcCodec)
+      .typecase(130, updateFulfillHtlcCodec)
+      .typecase(131, updateFailHtlcCodec)
+      .typecase(134, updateFeeCodec)
+      .typecase(135, updateFailMalformedHtlcCodec)
```

Comment:
Never noticed that those messages weren't following the _it's ok to be odd_ rule.

Reply:
Yes, these are fundamental messages that need to be understood by everyone.
I'm even surprised that the `UpdateFail*` messages are odd instead of being even...

Reply:
Right! They should all be even. Well...